ui.js

```diff
@@ -164,27 +169,34 @@ class Emoj extends Component {
 		this.fetchEmojis(query);
 	}
 
-	handleKeyPress(ch, key = {}) {
+	handleInput(data) {
 		const {onExit, onSelectEmoji} = this.props;
 		let {skinNumber, selectedIndex, emojis, query} = this.state;
 
-		// Filter out all ansi sequences except the up/down keys which change the skin tone
-		// and left/right keys which select emoji inside a list
-		const isArrowKey = ['up', 'down', 'left', 'right'].includes(key.name);
+		const s = String(data);
```

Comment:
Use a proper variable name instead of `s`.

Reply:
Instead of converting `data` to String each time, you should rather change the encoding of the stream: `this.props.stdin.setEncoding('utf8')`. This is both more efficient and also handles unicode better as it would never split multi-byte unicode characters. Maybe `StdinContext.Consumer` should have an `encoding` prop?

Reply:
Good point! I decided to set it to `utf8` always, because I couldn't figure out a way how to reset it back to emit `Buffer` instead of `string`, after setting encoding to `utf8`.